### PR TITLE
Implement flash messages with separate namespace.

### PR DESCRIPTION
**Pull request by pchapl**
Logic was ported from Yii 2, with the exception of separate namespace for flashes.

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #100 #31

### CODE REVIEW

**Comment by samdark** [comment]
@pchapl looks good. Could you add tests for this class?

File: src/Flash.php [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+use Yiisoft\Yii\Web\Session\SessionInterface;
+
+class Flash
+{
+    private const COUNTERS = '__counters';
+
+    /**
+     * @var string the name of the session variable that stores the flash message data.
+     */
+    public $flashParam = '__flash';
```
[the remark this answers is not in the document]
Comment (pchapl) [reply src/Flash.php]: Looks like it's very rare case. I've made it a const

File: src/Flash.php [hunk 2]
```diff
@@ -0,0 +1,232 @@
+<?php
+
+namespace Yiisoft\Yii\Web;
+
+use Yiisoft\Yii\Web\Session\SessionInterface;
+
+class Flash
```
[the remark this answers is not in the document]
Comment (pchapl) [reply src/Flash.php]: Yep

**Comment by rob006** [comment]
There should be an interface for this component.

**Comment by pchapl** [comment]
@rob006 You're right. Thanks.

**Comment by samdark** [comment]
@pchapl is it ready for review?

**Comment by pchapl** [comment]
Yes. I could not fully test due to problems with widgets in yii 3, but it's ready.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**